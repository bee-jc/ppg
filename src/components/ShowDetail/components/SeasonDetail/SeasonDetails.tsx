import React from 'react';

import { SeasonNumber, EpisodeCount } from '../../../../styles';

const SeasonDetails: React.FC<{
  seasonNumber: number;
  episodeCount: number;
}> = ({ seasonNumber, episodeCount }) => {
  const formattedNumber =
    seasonNumber < 10 ? `S0${seasonNumber}` : `S${seasonNumber}`;
  return (
    <>
      <SeasonNumber>{formattedNumber}</SeasonNumber>
      <EpisodeCount>({episodeCount} episodes)</EpisodeCount>
    </>
  );
};

export default SeasonDetails;
