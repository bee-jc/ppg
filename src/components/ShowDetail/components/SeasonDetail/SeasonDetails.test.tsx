import { render, screen } from '@testing-library/react';
import SeasonDetails from './SeasonDetails';

describe('SeasonDetails', () => {
  const seasonNumber = 1;
  const episodeCount = 10;

  it('should render the season number and episode count correctly', () => {
    const { asFragment } = render(
      <SeasonDetails seasonNumber={seasonNumber} episodeCount={episodeCount} />
    );

    const formattedNumber =
      seasonNumber < 10 ? `S0${seasonNumber}` : `S${seasonNumber}`;
    expect(screen.getByText(formattedNumber)).toBeInTheDocument();
    expect(screen.getByText(`(${episodeCount} episodes)`)).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
