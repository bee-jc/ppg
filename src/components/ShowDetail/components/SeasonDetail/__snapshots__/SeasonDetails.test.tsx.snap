// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeasonDetails should render the season number and episode count correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1vdtga9"
  >
    S01
  </div>
  <div
    class="css-1wqfxc5"
  >
    (10 episodes)
  </div>
</DocumentFragment>
`;
